### PR TITLE
feat: WebLn and Nostr providers communicating with front-end

### PR DESCRIPTION
**Pull request by drablittlecrab**
## Description
Providers communicating with front-end
- WebLN
- Nostr

PubKey
- Automatically populating with use of Nostr

Radio Buttons Disabled
- When Provider loading
- When provider not connected

## TODO
- Implement testing
- Connect to server for registration

### CODE REVIEW

File: src/utils/nostr.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
+/* eslint-disable react-refresh/only-export-components */
+import NDK, { NDKNip07Signer, NDKUser } from "@nostr-dev-kit/ndk";
+import { createContext, useContext, useEffect, useState } from "react";
+
+export const RELAYS = ["wss://relay.snort.social", "wss://nos.lol"];
```
Comment (Kodylow) [inline src/utils/nostr.tsx]: should set these in config with env vars

File: src/utils/nostr.tsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+          "nostr" in window &&
+          window.nostr &&
+          (("isEnabled" in window.nostr &&
+            typeof window.nostr.isEnabled === "function" &&
+            (await window.nostr?.isEnabled())) ||
+            ("isEnabled" in window.nostr &&
+              typeof window.nostr.isEnabled === "boolean" &&
+              window.nostr.isEnabled) ||
+            ("_isEnabled" in window.nostr && window.nostr?._isEnabled))
+        ) {
```
Comment (Kodylow) [inline src/utils/nostr.tsx]: you don't need all these typechecks, just define the window.nostr global interface somewhere and assert that window.nostr exists

**Review by Kodylow — APPROVED** [review]
looks great, minor nits

**Comment by drablittlecrab** [comment]
Sorry for the rapid-fire commits...

**Pull request merged**